src/readbin/symbols.ml

```diff
@@ -66,4 +66,22 @@ let read ?demangle ~filename arch base  : string table =
             let mem = Memory.view ~from ~words base |> ok_exn in
             let name = demangle name in
             Table.add syms mem name |> ok_exn
-          with exn -> syms))
+          with _exn -> syms))
+
+let read_addrset filename : Addr.Hash_set.t =
+  let ic = In_channel.create filename in
+  let set_string = In_channel.input_all ic in
+  Addr.Hash_set.t_of_sexp @@ Sexp.of_string set_string
+
+let write_addrset ?filename (addr_set : Addr.Hash_set.t) : unit =
+  let oc = match filename with
+    | None -> Out_channel.stdout
+    | Some f -> Out_channel.create f in
+  Out_channel.output_string oc @@ Sexp.to_string (Addr.Hash_set.sexp_of_t addr_set);
+  Out_channel.close oc
+
+let write ?filename (syms : symbol table) : unit =
+  let fs_l = Table.foldi syms ~init:[] ~f:(fun mem _sym fs_list ->
```

Comment:
I believe I changed it in the later request. :) I found I did a lot Table
-> list -> set before, which is not necessary. I've changed them. Thank you
for pointing out!

On Sun, Apr 19, 2015 at 4:24 PM Ivan Gotovchits <notifications@github.com>
wrote:

> In src/readbin/symbols.ml
> <https://github.com/BinaryAnalysisPlatform/bap/pull/182#discussion_r28656351>
> :
>
> > +          with _exn -> syms))
> > +
> > +let read_addrset filename : Addr.Hash_set.t =
> > +  let ic = In_channel.create filename in
> > +  let set_string = In_channel.input_all ic in
> > +  Addr.Hash_set.t_of_sexp @@ Sexp.of_string set_string
> > +
> > +let write_addrset ?filename (addr_set : Addr.Hash_set.t) : unit =
> > +  let oc = match filename with
> > +    | None -> Out_channel.stdout
> > +    | Some f -> Out_channel.create f in
> > +  Out_channel.output_string oc @@ Sexp.to_string (Addr.Hash_set.sexp_of_t addr_set);
> > +  Out_channel.close oc
> > +
> > +let write ?filename (syms : symbol table) : unit =
> > +  let fs_l = Table.foldi syms ~init:[] ~f:(fun mem _sym fs_list ->
>
> That's ok, but if you want set, why you're using list? ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BinaryAnalysisPlatform/bap/pull/182/files#r28656351>.
>